### ISSUE TITLE
Peer Code Review

### ISSUE THREAD

**Issue by kaavyakalani26** [sainanim/tutorial2#1]
```
---
title: "tut2"
format: pdf
editor: visual
---
```
To begin with, the title of your file can be more focused on what the code in the file is about rather than this, something like "Making a graph using a dataset from opendata toronto".

Additionally more fields can be added to the top matter, some of the essential ones being author and date. This is important for credibility and documentation purposes, for the reader to clearly know who wrote this code and when.

Additionally, adding a Bibliography using a bib file is also a good practice to be included for references and then some citations it could include in this case would be for R and the libraries used.

```{r}
#| echo: false
#| message: false
#| warning: false

# install.packages(c("tidyverse", "opendatatoronto"))
# install.packages("janitor")
# install.packages("knitr")

library(opendatatoronto)
library(tidyverse)
library(janitor)
library(knitr)
```
You have started off great by installing all the necessary packages for your analysis and then calling them to be used. Commenting out the install packages call is also great as you don't need to execute that again after executing it once but if an external person loos at your file, they exactly know what and how to install.

```
#### Acquire ####
raw_igme_data <-
  read_csv(
    file =
      "https://childmortality.org/wp-content/uploads/2021/09/UNIGME-2021.csv",
    show_col_types = FALSE
  )

write_csv(x = raw_igme_data, file = "igme.csv")
head(raw_igme_data)

names(raw_igme_data)

```
The next step you do is to pick a dataset from the Open Data Toronto package and import it into R. Your code reads directly from a csv and I am unsure if this is from the Open Data Toronto package? This involved using a specific ID for the dataset in the package and then fetching that dataset.

Adding comments to your approach here would be really useful as it can help us understand where and how you get this dataset. If this is something manually added to your folder then you should also add it to your repository to ensure reproducibility.

```{r}
cleaned_igme_data <-
  clean_names(raw_igme_data) |>
  filter(
    sex == "Total",
    series_name == "UN IGME estimate",
    geographic_area %in% c("Argentina", "Australia", "Canada", "Kenya"),
    indicator == "Neonatal mortality rate") |>
  select(geographic_area, time_period, obs_value)

head(cleaned_igme_data)

cleaned_igme_data <-
  cleaned_igme_data |>
  mutate(
    time_period = str_remove(time_period, "-06"),
    time_period = as.integer(time_period)
  ) |>
  filter(time_period >= 1971) |>
  rename(nmr = obs_value, year = time_period, country = geographic_area)

head(cleaned_igme_data)
```

```{r}
write_csv(x = cleaned_igme_data, file = "cleaned_igme_data.csv")

#### Explore ####
cleaned_igme_data <-
  read_csv(
    file = "cleaned_igme_data.csv",
    show_col_types = FALSE
  )

head(cleaned_igme_data)
```
You do a great extensive job of cleaning the data and writing it to a new file which you will later use to create the graph. Again, adding comments to document what you are doing is very essential as it not only helps you come back to your code at a later date  and know what you were doing but also helps an another person reading your code know exactly what you you were doing - a main aim of repoducibility. It helps you track what decisions you take at every step and why. 

Using head() at every step to display part of the content is a great practice so that the user has an idea of the dataset/ dataset version you are currently working with and helps give context to any discussions you do or any plots you make.  

```{r}
cleaned_igme_data |>
  ggplot(aes(x = year, y = nmr, color = country)) +
  geom_point() +
  theme_minimal() 
```
Your graph looks great and the type of graph is appropriately chosen for the data you are trying to represent graphically. Adding informative labels to the axis and a title to the graph is something which is missing here and should be added. Adding those labels helps the reader to clearly understand what your graph is trying to convey. For example the y axis currently saying "nmr" might not be the most reader friendly as not everyone necessarily knows what it stands for. Using a label such as "Neonatal Mortality Rate"is a better label. Additionally a title of the graph helps the reader get a overview of what the graph is trying to showcase before they get into the details of the plot. Adding comments here could be useful as well.

In conclusion, you do a great job of cleaning the chosen data set, choosing an appropriate graph form and plotting the graph. Enhancing the report with better documentation, especially in the form of comments, would make this report even better and reproducible to a higher degree.
